### ISSUE TITLE
The example script still does not reproduce the results

### ISSUE THREAD

**Issue by randomboolean** [danielegrattarola/keras-gat#17]
I have run the example script 100 times,
I obtained a mean accuracy of 81.54% with standard deviation 0.75%.
This is close from GCN results (81.5%), but far from the 83.0% that was reported in GAT paper.

Simulations were done with the latest commit 6e25df0fa8f81691277114808d1ab1857fb4bfea

**Comment by danielegrattarola** [comment]
Thanks for pointing it out. 
Have you tried running [the official Tensorflow version by the authors]() and comparing results? 
I'm looking over the code right now and everything seems implemented properly, so I wouldn't know where to look for the issue. 
I'm also noticing that in the GAT paper they don't report the standard dev, so maybe 83% is the best result?

**Comment by danielegrattarola** [comment]
Ok try it now, commit 95b0c30 should fix the issue.

**Comment by randomboolean** [comment]
I've tried it on commit 26485f63e2a51ee9489199a6a182a1a717457d4a.
I obtained 81.70 +- 0.79 % on 100 runs, so that didn't solve the issue.
I thought it might be because the metric that was monitored to save the model checkpoint wasn't the right one, so I added the flag `monitor='val_weighted_acc'` in `ModelCheckpoint`, but I obtained worse (80.83 +- 1.27 %).
For comparison with GAT official repo I obtained 83.25 +- 0.59 %. But when I remove the lines where it saves the best model, it drops to 81.70 +- 0.79 %.

**Comment by randomboolean** [comment]
FYI, following your suggestion, I had run a pure GCN (16 hidden units) with the additional dropout (and bias) and obtained 82.9 +- 0.7 (and by using the model with the best val acc for test).
Also adding dropout on the adjacency matrix (which is equivalent to apply dropout on the attention coefficients in GAT) raised the test accuracy to 83.4 +- 0.7
